### PR TITLE
Split single ufuncs.c file in multiple files

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
This is an attempt to split the single (big) ufuncs C file in multiple files.

I find this a bit easier to understand the code if it is not all in a huge file, but it was also just a good exercise to get familiar with the code, so @caspervdw you certainly don't need to like this ;)

### CODE REVIEW

**Comment by caspervdw** [comment]
This is great, thanks for doing this work. I am surprised AppVeyor isn't picking up this PR, probably I switched PR building off for some reason.

File: src/pygeom.c [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+    .tp_members = GeometryObject_members,
+    .tp_methods = GeometryObject_methods,
+    .tp_repr = (reprfunc) GeometryObject_repr,
+};
+
+int
+init_geom_type(PyObject *m)
+{
+    if (PyType_Ready(&GeometryType) < 0)
+        return NULL;
```
Comment (jorisvandenbossche) [inline src/pygeom.c]: There is one remaining (newly introduced) build warning I didn't fully solve. It is complaining here that it is returning NULL (a pointer), while I declared it to return `int`. And also that the rest does not return anything.

But maybe I can return here 0 / -1, and do the `return NULL` in the actual `PyInit_ufuncs` function

Comment (caspervdw) [reply src/pygeom.c]: I think you should indeed return 0 / -1 and depending on that, `return NULL` in the `PyInit_ufuncs` (in `pygeos/ufuncs.c`) if necessary

File: src/pygeom.c [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+        return GeometryObject_FromWKB(type, value);
+    }
+    else if (PyUnicode_Check(value)) {
+        return GeometryObject_FromWKT(type, value);
+    }
+    else {
+        PyErr_Format(PyExc_TypeError, "Expected string or bytes, found %s", value->ob_type->tp_name);
+        return NULL;
+    }
+    ptr = GEOSGeom_clone_r(context_handle, arg);
```
Comment (caspervdw) [inline src/pygeom.c]: This code is unreachable! Should be deleted. My bad, I think

File: src/pygeom.c [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+    .tp_dealloc = (destructor) GeometryObject_dealloc,
+    .tp_members = GeometryObject_members,
+    .tp_methods = GeometryObject_methods,
+    .tp_repr = (reprfunc) GeometryObject_repr,
+};
+
+int
+init_geom_type(PyObject *m)
+{
+    if (PyType_Ready(&GeometryType) < 0)
```
Comment (caspervdw) [inline src/pygeom.c]: I think the normal do this kind of check is `if (!PyType_Ready(&GeometryType))`

**Comment by caspervdw** [comment]
I now realize that all feedback is related to stuff I wrote myself. Feel free to merge as is (except for the compiler warning, would be nice to solve that)

**Comment by jorisvandenbossche** [comment]
No problem, I can address those at once when fixing the other issue.

One additional thought I had is if we want to keep the "ufuncs" name for the compiled module in Python? Since it already contains more as just the ufuncs (it also already contains the base Geometry type), and might contain more in the future (dtype, STRTree, other algos like a join). 
Eg, we could have a single compiled module `pygeos.lib`.

**Comment by jorisvandenbossche** [comment]
Or, we could actually also expose them as multiple python modules, although I don't know how that works with interlinking between them on the C level if they are not compiled into a single linked module.

Anyway, in the end, users don't directly use this but rather the wrapped python functions, so the name is also not that important.

File: src/pygeom.c (hunk 3 above)
Comment (jorisvandenbossche) [reply src/pygeom.c]: Here this is actually correct, as `PyType_Ready` does not return NULL on failure but -1

**Comment by caspervdw** [comment]
> One additional thought I had is if we want to keep the "ufuncs" name for the compiled module in Python? Since it already contains more as just the ufuncs (it also already contains the base Geometry type), and might contain more in the future (dtype, STRTree, other algos like a join).
Eg, we could have a single compiled module pygeos.lib.

I like the idea of making this more modular on the C level, and expose it with `pygeos.lib` as a single module. Something for a next round of refactor.

**Pull request merged**